Security/Guidelines/Web_Security.mediawiki

```diff
@@ -0,0 +1,649 @@
+|- style="background-color: #D2D2D2;"
+| data-sort-value="11" | [[#CSRF Prevention|<span style="color: black;">Cross-site Request Forgery Tokenization</span>]]
+| style="text-align: center;" | P2
+| style="text-align: center;" | Varies
+| style="text-align: center;" | 6
+| Varies
+| Mandatory for websites that allow destructive changes<br>Unnecessary for all other websites<br>Most application frameworks have built-in CSRF tokenization to ease implementation
+|- style="background-color: #D2D2D2;"
+| data-sort-value="12" | [[#robots.txt|<span style="color: black;">robots.txt</span>]]
+| style="text-align: center;" | P5
+| style="text-align: center;" | Easy
+| style="text-align: center;" | 13
+| Optional
+| Websites that implement robots.txt must use it only for noted purposes
+|- style="background-color: #E8E27A;"
+| data-sort-value="13" | [[#Subresource Integrity|<span style="color: black;">Subresource Integrity</span>]]
+| style="text-align: center;" | P5
+| style="text-align: center;" | Moderate
+| style="text-align: center;" | 14
+| Recommended<sup style="font-size: .8em; position: relative; top: -.4em; vertical-align: baseline;">&Dagger;</sup>
+| <sup style="font-size: .8em; position: relative; top: -.4em; vertical-align: baseline;">&Dagger;</sup> Only for websites that load JavaScript or stylesheets from foreign origins
+|- style="background-color: #E8E27A;"
+| data-sort-value="14" | [[#X-Content-Type-Options|<span style="color: black;">X-Content-Type-Options</span>]]
+| style="text-align: center;" | P3
+| style="text-align: center;" | Easy
+| style="text-align: center;" | 8
+| Recommended for all websites
+| Websites should verify that they are setting the proper MIME types for all resources
+|- style="background-color: #9EDB58;"
+| data-sort-value="15" | [[#X-Frame-Options|<span style="color: black;">X-Frame-Options</span>]]
+| style="text-align: center;" | P2
+| style="text-align: center;" | Easy
+| style="text-align: center;" | 5
+| Mandatory for all websites
+| Websites that don't use DENY or SAMEORIGIN must employ clickjacking defenses
+|- style="background-color: #E8E27A;"
+| data-sort-value="16" | [[#X-XSS-Protection|<span style="color: black;">X-XSS-Protection</span>]]
+| style="text-align: center;" | P4
+| style="text-align: center;" | Moderate
+| style="text-align: center;" | 12
+| Mandatory for all new websites<br>Recommended for existing websites
+| Manual testing should be done for existing websites, prior to implementation
+|}
+
+<div style="margin-left: 1.5em;"><sup style="font-size: .8em; position: relative; top: -.4em; vertical-align: baseline;">&dagger;</sup> Suggested order that administrators implement the web security guidelines. It is based on a combination of the security impact and the ease of implementation from an operational and developmental perspective.</div>
+
+
+
+<div style="max-width: 75em;">
+= Transport Layer Security =
```

Comment:
Perhaps this section could be called "Transport Layer Security (TLS/SSL)" to match the first five words of line 182 under HTTPS, below - and to help people who recognize "SSL" but *not* the full expanded TLS acronym recognize it in the ToC?